examples/server_backup/main.go

```diff
@@ -0,0 +1,387 @@
+					Description: "The channel to backup",
+					Required:    true,
+					ChannelTypes: []discordgo.ChannelType{
+						discordgo.ChannelTypeGuildText,
+					},
+				},
+			},
+		},
+		{
+			Name:                     "list-backups",
+			Description:              "List all available backups",
+			DefaultMemberPermissions: &defaultMemberPermissions,
+			DMPermission:             &dmPermission,
+		},
+		{
+			Name:                     "get-backup",
+			Description:              "Get a specific backup file",
+			DefaultMemberPermissions: &defaultMemberPermissions,
+			DMPermission:             &dmPermission,
+			Options: []*discordgo.ApplicationCommandOption{
+				{
+					Type:        discordgo.ApplicationCommandOptionString,
+					Name:        "filename",
+					Description: "The backup file to retrieve",
+					Required:    true,
+				},
+			},
+		},
+	}
+
+	commandHandlers = map[string]func(s *discordgo.Session, i *discordgo.InteractionCreate){
+		"backup-server": func(s *discordgo.Session, i *discordgo.InteractionCreate) {
+			// Acknowledge the command immediately
+			s.InteractionRespond(i.Interaction, &discordgo.InteractionResponse{
+				Type: discordgo.InteractionResponseDeferredChannelMessageWithSource,
+				Data: &discordgo.InteractionResponseData{
+					Content: "Starting server backup...",
+				},
+			})
+
+			guild, err := s.Guild(i.GuildID)
+			if err != nil {
+				sendFollowUpError(s, i, "Failed to get guild information", err)
+				return
+			}
+
+			backup := &BackupData{
+				GuildID:    guild.ID,
+				GuildName:  guild.Name,
+				BackupTime: time.Now(),
+				Channels:   make([]*ChannelBackup, 0),
+			}
+
+			channels, err := s.GuildChannels(i.GuildID)
+			if err != nil {
+				sendFollowUpError(s, i, "Failed to get guild channels", err)
+				return
+			}
+
+			for _, channel := range channels {
+				if channel.Type != discordgo.ChannelTypeGuildText {
+					continue
+				}
+
+				messages, err := FetchChannelHistory(s, channel.ID)
+				if err != nil {
+					log.Printf("Error backing up channel %s: %v", channel.Name, err)
+					continue
+				}
+
+				backup.Channels = append(backup.Channels, &ChannelBackup{
+					ID:       channel.ID,
+					Name:     channel.Name,
+					Type:     channel.Type,
+					Messages: messages,
+				})
+			}
+
+			// Create backup directory if it doesn't exist
+			if err := os.MkdirAll(*BackupDir, 0755); err != nil {
+				sendFollowUpError(s, i, "Failed to create backup directory", err)
+				return
+			}
+
+			// Save backup to file
+			filename := fmt.Sprintf("%s/%s_%s.json", *BackupDir, guild.Name, time.Now().Format("2006-01-02_15-04-05"))
+			data, err := json.MarshalIndent(backup, "", "  ")
+			if err != nil {
+				sendFollowUpError(s, i, "Failed to marshal backup data", err)
+				return
+			}
+
+			if err := os.WriteFile(filename, data, 0644); err != nil {
+				sendFollowUpError(s, i, "Failed to write backup file", err)
+				return
+			}
+
+			s.InteractionResponseEdit(i.Interaction, &discordgo.WebhookEdit{
```

Comment:
isn't this going to timeout (interaction failed) while collecting all that data?